File: ios/dayce/Logic/DayCalculator.swift
import Foundation

enum DayCalculator {
    static func dayValue(
        for eventDate: Date,
        kind: DayEventKind,
        now: Date = .now,
        calendar: Calendar = .current
    ) -> Int {
        let startOfEvent = calendar.startOfDay(for: eventDate)
        let startOfNow = calendar.startOfDay(for: now)
        let days = calendar.dateComponents([.day], from: startOfNow, to: startOfEvent).day ?? 0

        switch kind {
        case .countUp:
            return -days + 1
        case .countDown:
            return days
        }
    }

    static func suffix(for kind: DayEventKind, value: Int) -> String {
        switch kind {
        case .countUp:
            return value == 1 ? "day" : "days"
        case .countDown:
            if value > 0 {
                return value == 1 ? "day left" : "days left"
            }
            if value == 0 {
                return "today"
            }
            return abs(value) == 1 ? "day ago" : "days ago"
        }
    }
}

File: android/app/src/main/java/com/motoish/dayce/domain/DayCalculator.kt
package com.motoish.dayce.domain

import java.time.LocalDate
import java.time.temporal.ChronoUnit

object DayCalculator {
    fun dayCount(kind: DayEventKind, eventDate: LocalDate, today: LocalDate): Long {
        return when (kind) {
            DayEventKind.CountUp -> {
                val days = ChronoUnit.DAYS.between(eventDate, today) + 1
                days.coerceAtLeast(0)
            }
            DayEventKind.Countdown -> ChronoUnit.DAYS.between(today, eventDate)
        }
    }
}
